iwyu_tool.py

```diff
@@ -25,9 +25,67 @@
 import json
 import argparse
 import subprocess
-
-
-def run_iwyu(cwd, compile_command, iwyu_args, verbose):
+import re
+
+
+CORRECT_RE = re.compile(r'^\((.*?) has correct #includes/fwd-decls\)$')
+SHOULD_ADD_RE = re.compile(r'^(.*?) should add these lines:$')
+SHOULD_REMOVE_RE = re.compile(r'^(.*?) should remove these lines:$')
+FULL_LIST_RE = re.compile(r'The full include-list for (.*?):$')
+END_RE = re.compile(r'^---$')
+LINES_RE = re.compile(r'.*// lines ([0-9]+)-[0-9]+$')
+
+
+GENERAL, ADD, REMOVE, LIST = range(4)
```

Comment:
iwyu_test_util.py says:

    This script has been tested with python 2.7, 3.1.3 and 3.2.

So I think 2.7 and later is fine. I didn't even know real enums were available in Python, but 3.4 is probably a little too harsh a requirement. `namedtuple` sounds good to me!